File: serialization/tabular/databank/CSharp/DatabankParserBase.cs
using System;
using System.Collections.Generic;
using System.IO;
using Antlr4.Runtime;

public abstract class DatabankParserBase : Parser
{
    protected DatabankParserBase(ITokenStream input)
        : base(input)
    {
    }

    protected DatabankParserBase(ITokenStream input, TextWriter output, TextWriter errorOutput)
        : base(input, output, errorOutput)
    {
    }


    protected bool isdatatype()
    {
        var text = ((ITokenStream)this.InputStream).LT(-1).Text;
        return text == "-1" || text == "-4" || text == "-12";
    }
}

File: serialization/tabular/databank/Java/DatabankParserBase.java
import java.util.List;
import org.antlr.v4.runtime.*;

/**
 * All parser methods that used in grammar (p, prev, notLineTerminator, etc.)
 * should start with lower case char similar to parser rules.
 */
public abstract class DatabankParserBase extends Parser
{
    protected DatabankParserBase(TokenStream input) {
        super(input);
    }


    /**
     * Returns true if the current Token is a closing bracket (")" or "}")
     */
    protected boolean isdatatype()
    {
	String text = _input.LT(-1).getText();
	return text.equals("-1") || text.equals("-4") || text.equals("-12");
    }
}
